unit/util/replace_symbol.cpp

```diff
@@ -0,0 +1,61 @@
+/*******************************************************************\
+
+ Module: replace_symbolt unit tests
+
+ Author: Michael Tautschnig
+
+\*******************************************************************/
+
+#include <testing-utils/catch.hpp>
+
+#include <util/replace_symbol.h>
+#include <util/std_expr.h>
+
+TEST_CASE("Replace all symbols in expression", "[core][util][replace_symbol]")
+{
+  symbol_exprt s1("a", typet("some_type"));
+  symbol_exprt s2("b", typet("some_type"));
+
+  exprt binary("binary", typet("some_type"));
+  binary.copy_to_operands(s1);
+  binary.copy_to_operands(s2);
+
+  exprt other_expr("other");
+
+  replace_symbolt r;
+  REQUIRE(r.empty());
+
+  r.insert("a", other_expr);
+  REQUIRE(r.replaces_symbol("a"));
+  REQUIRE(r.get_expr_map().size() == 1);
+
+  REQUIRE(r.replace(binary) == false);
+  REQUIRE(binary.op0() == other_expr);
+
+  REQUIRE(r.replace(s1) == false);
+  REQUIRE(s1 == other_expr);
+
+  REQUIRE(r.replace(s2) == true);
+  REQUIRE(s2 == symbol_exprt("b", typet("some_type")));
+
+  REQUIRE(r.erase("a") == 1);
+  REQUIRE(r.empty());
+}
+
+TEST_CASE("Lvalue only", "[core][util][optional]")
```

Comment:
Weird name / category. Surely this is a [replace_symbol] test like above?

Reply:
Thank you for catching this! I guess it tells you which copy&paste I started from ...

Reply:
Fixed in #2724 (where this commit primarily lives).